src/otk_external_osbuild/command/gen_depsolve_dnf4.py

```diff
@@ -16,6 +16,14 @@ def transform(packages):
     # other externals.
     data["tree"]["const"]["internal"]["packages"] = packages
 
+    # We also store all resolved packages and some meta information about
+    # them, this just turns the list into a more user-friendly accessible
+    # map keyed by package name.
+    data["tree"]["const"]["versions"] = {}
+
+    for package in packages:
+        data["tree"]["const"]["versions"][package["name"]] = package
```

Comment:
If we do it like this then the internal json representation of a "package" in dnfjson becomes an (external) API that we will need to maintain, i.e. we can never rename/remove/change e.g. `remote_location` or `checksum`. I would suggest to change it so that we only expose what is required for the kernel use-case instead as this is easier to maintain/keep compatible.

Fwiw, this is orthogonal to my point that I'm not sure we should expose the details for all packages by default (but limiting what we expose would certainly reduce it)